File: src/main/java/me/miki/shindo/management/settings/metadata/SettingMetadata.java
package me.miki.shindo.management.settings.metadata;

import lombok.Getter;
import lombok.Setter;

@Getter
public class SettingMetadata {
    private final String fieldName;

    @Setter
    private String category = "";

    @Setter
    private String description = "";

    @Setter
    private String keyOverride = "";

    @Setter
    private boolean hidden = false;

    @Setter
    private double min = Double.NaN;

    @Setter
    private double max = Double.NaN;

    @Setter
    private double step = Double.NaN;

    public SettingMetadata(String fieldName) {
        this.fieldName = fieldName;
    }
}

File: src/main/kotlin/me/miki/shindo/management/settings/metadata/SettingMetadata.kt
package me.miki.shindo.management.settings.metadata

class SettingMetadata(
    val fieldName: String,
) {
    var category: String = ""
    var description: String = ""
    var keyOverride: String = ""
    var hidden: Boolean = false
    var min: Double = Double.NaN
    var max: Double = Double.NaN
    var step: Double = Double.NaN
}
